lib/net/validations.rb

```diff
@@ -119,6 +119,13 @@ def self.normalize_ip6(ip)
       IPAddr.new(ip, Socket::AF_INET6).to_s rescue ip
     end
 
+    def self.normalize_network_address(ip, mask)
+      IPAddr.new(ip).mask(mask).to_s
+    rescue StandardError => e
+      Foreman::Logging.logger('app').info "Unable to validate network address: #{e}"
```

Comment:
The method is called normalize, not validate, so the message seems wrong?

Also I wonder under which circumstances we can get an error and yet still want to continue using that address?

Reply:
The method needs to be called via `before_validate` so it is normalized early in the callback so orchestration already has it in the correct format. There are two options:

1) Raise an exception which will render to the ugly 500 page,
2) Log an error and return something. That is the path I have chosen, I do believe there will only be exception when something got utterly wrong, for example netmask does not make sense or the input is not a valid IP address which will be caught later on via `ensure_ip_addrs_valid`